src/dispatch/plugins/dispatch_slack/incident/interactive.py

```diff
@@ -2097,3 +2101,70 @@ async def handle_report_incident_project_select_action(
         trigger_id=body["trigger_id"],
         view=modal,
     )
+
+
+# BUTTONS
+
+
+@app.action(
+    IncidentNotificationActions.invite_user, middleware=[button_context_middleware, db_middleware]
+)
+async def handle_incident_notification_join_button_click(
+    ack: AsyncAck,
+    body: dict,
+    client: AsyncWebClient,
+    respond: AsyncRespond,
+    db_session: Session,
+    context: AsyncBoltContext,
+):
+    """Handles the incident join button click event."""
+    await ack()
+    incident = incident_service.get(db_session=db_session, incident_id=context["subject"].id)
+
+    if not incident:
+        message = "Sorry, we can't invite you to this incident. The incident does not exist."
+    elif incident.visibility == Visibility.restricted:
+        message = "Sorry, we can't invite you to this incident. The incident's visbility is restricted. Please, reach out to the incident commander if you have any questions."
+    elif incident.status == IncidentStatus.closed:
+        message = "Sorry, you can't join this incident. The incident has already been marked as closed. Please, reach out to the incident commander if you have any questions."
+    else:
+        user_id = context["user_id"]
+        await dispatch_slack_service.add_users_to_conversation_async(
```

Comment:
I don't think we need this function. You could call `client.conversations_invite().`